src/modules/shortcut.ts

```diff
@@ -0,0 +1,202 @@
+
+	@command({
+		single: true,
+		description: 'Shortcut: Create or edit a shortcut',
+	})
+	async shortcut(msg: Message, arg: string) {
+		const [name, ...titleParts] = arg.split(' ');
+		const title = titleParts?.join(' ') ?? '';
+
+		if (!name) {
+			return await sendWithMessageOwnership(
+				msg,
+				':x: You have to supply a name for the command',
+			);
+		}
+
+		const id = this.getId(msg.author, name);
+		const existingShortcut = await this.getShortcut(id);
+
+		if (existingShortcut && existingShortcut.owner !== msg.author.id)
+			return await sendWithMessageOwnership(
+				msg,
+				":x: Cannot edit another's shortcut",
+			);
+
+		const referencedMessage = await msg.channel.messages.fetch(
+			msg.reference?.messageID!,
+		);
+		if (!referencedMessage)
+			return await sendWithMessageOwnership(
+				msg,
+				':x: You have to reply to a comment to make it a shortcut',
+			);
+
+		const description = referencedMessage.content;
+		const referencedEmbed = referencedMessage.embeds[0];
+		const base = {
+			id,
+			uses: existingShortcut?.uses ?? 0,
+			owner: msg.author.id,
+		};
+
+		let data: Omit<Shortcut, keyof BaseEntity> | undefined;
+		if (LINK_REGEX.exec(description)?.[0] === description)
+			data = {
+				...base,
+				content: description,
+			};
+		else if (description)
+			data = {
+				...base,
+				title,
+				description,
+				color: BLOCKQUOTE_GREY_NUMBER,
+			};
+		else if (referencedEmbed)
+			data = {
+				...base,
+				title:
+					referencedEmbed.title && title
+						? `${title}: ${referencedEmbed.title}`
+						: referencedEmbed.title || title,
+				description: referencedEmbed.description,
+				color: referencedEmbed.color,
+				image: referencedEmbed.image?.url,
+				url: referencedEmbed.url,
+			};
+
+		if (!data)
+			return await sendWithMessageOwnership(
+				msg,
+				':x: Cannot generate a shortcut from that message',
+			);
+
+		await existingShortcut?.remove();
+		await Shortcut.create(data).save();
+		await sendWithMessageOwnership(
+			msg,
+			`:white_check_mark: ${
+				existingShortcut ? 'Edited' : 'Created'
+			} shortcut \`${id}\``,
+		);
+	}
+
+	private async getShortcut(id: string) {
+		return await Shortcut.findOne(id);
+	}
+
+	@command({
+		aliases: ['ls'],
+		description: 'Shortcut: List the most used shortcuts',
+	})
+	async listShortcuts(msg: Message) {
+		const shortcuts: Shortcut[] = await Shortcut.createQueryBuilder()
+			.select(['id', 'uses'])
+			.orderBy('uses', 'DESC')
+			.limit(10)
+			.getRawMany();
+		await sendWithMessageOwnership(msg, {
+			embed: new MessageEmbed()
+				.setTitle('Top 10 Shortcuts')
+				.setColor(TS_BLUE)
+				.setDescription(
+					shortcuts.map(
+						shortcut =>
+							`- \`${shortcut.id}\` with **${shortcut.uses}** uses.`,
+					),
+				),
+		});
+	}
+
+	@command({
+		aliases: ['lsm'],
+		description: 'Shortcut: List shortcuts owned by you',
+	})
+	async listMyShortcuts(msg: Message) {
+		const shortcuts: Shortcut[] = await Shortcut.createQueryBuilder()
```

Comment:
I don't really remember the specifics with TypeORM but I think you might be able to get away with no QueryBuilder.